Sage/carbon
# fix(form): update onSubmit type

fix #5659

### Proposed behaviour


Updates the `onSubmit` type so that the `Form` component can be used in conjunction with external libraries such as `Formik`

### Current behaviour


Type for `onSubmit` means that the `Form` component is no longer compatible with libraries like Formik

### Checklist




#### QA


### Additional context



### Testing instructions


Linked sandbox below should have no type error thrown but the one from the linked issue will etc

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
